apps/prairielearn/src/migrations/20250917191818__course_instances__rename_requires_secret_link.sql

```diff
@@ -0,0 +1,5 @@
+ALTER TABLE course_instances
+DROP COLUMN self_enrollment_requires_secret_link;
```

Comment:
Super duper unsafe. What happens when a server running older code does `SELECT * FROM course_instances` and validates the result against `CourseInstanceSchema`?

You need to do this in multiple parts to do this correctly:

- Add the new column, update all writes to write to both the new and old columns.
- Update all code to read only the new column.
- Update all code to not write the old column.
- Drop the column.

Because this code more or less isn't actually being used yet (more precisely, we never read/write meaningful values from/to this column), you should be safe to combine the first three parts. But it is extremely unsafe to drop a column while there is still code out there that expects it to exist.